### PR TITLE
Adding Hive connectors

### PR DESCRIPTION
**Pull request by rsaidane**

### CODE REVIEW

File: HbaseOrchestration/resources/catalog/Phoenix_Hbase_Sql_Queries.xml [hunk 1]
```diff
@@ -102,75 +113,29 @@ def hbase_hdfs_site_config = "sed -i 's/HDFS-replacement/" + hdfs_path + "/g' \$
...
-upsert into test values (1,'Hello');
-upsert into test values (2,'World!');
-select * from test;"""
-
-
-
-
-
-
+//cmd.execute().waitForProcessOutput(System.out, System.err)
```
Comment (fviale) [inline HbaseOrchestration/resources/catalog/Phoenix_Hbase_Sql_Queries.xml:122]: remove this commented code

File: Hive/resources/catalog/Hive.xml [hunk 2]
```diff
@@ -90,6 +100,14 @@ def yarn_master_container_name = variables.get("yarn_master_container_name")
 def yarn_master_and_workers_pa_node_names = variables.get("yarn_master_and_workers_pa_node_names")
 def yarn_master_pa_node_name = yarn_master_and_workers_pa_node_names.split(",")[0]
 def targeted_network_name = hdfs_network_name
+def username = variables.get("username")
+def password = variables.get("password")
+
+if (username.isEmpty() || password.isEmpty()) {
```
Comment (fviale) [inline Hive/resources/catalog/Hive.xml:106]: this is not how you should set default variable values

File: Hive/resources/catalog/Hive.xml [hunk 3]
```diff
@@ -127,7 +145,7 @@ while (!docker_exec_control_output.toString().contains("mysqld: ready for connec
 
 // Start  hive  container
 def hive_container_name = instance_name + "-hive"
-cmd = ["docker", "run", "--rm","-v", localspace + ":" + localspace, "-dit", "--publish", hive_gui_port + ":" + 9999, "-p", hiveServer2_gui_port + ":" + 10002, "--name", hive_container_name, "--net", targeted_network_name, "activeeon/hive:latest"]
+cmd = ["docker", "run", "--rm", "-dit", "--publish", hive_gui_port + ":" + 10002, "-p", hiveServer2_port + ":" + 10000, "--name", hive_container_name, "--net", targeted_network_name, "activeeon/hive:2"]
```
Comment (fviale) [inline Hive/resources/catalog/Hive.xml:148]: the fact that these ports are hardcoded prevent having 2 hive deployments on the same machine

File: Hive/resources/catalog/Hive.xml [hunk 4]
```diff
@@ -186,15 +205,16 @@ println cmd
 cmd.execute().waitForProcessOutput(System.out, System.err)
 
 // Propagate variables
-variables.put("hive_gui_port", hiveServer2_gui_port)
+variables.put("hive_gui_port", hive_gui_port)
 variables.put("targeted_network_name", targeted_network_name)
 variables.put("hive_container_name", hive_container_name)
-variables.put("hive_url", "hive://" + hive_container_name + ":10002")
+variables.put("hive_url", "hive://trydev2.activeeon.com:" + hiveServer2_port)
```
Comment (fviale) [inline Hive/resources/catalog/Hive.xml:211]: we should have a better default value e.g. localhost

File: Hive/resources/catalog/Hive.xml [hunk 5]
```diff
@@ -186,15 +205,16 @@ println cmd
...
 variables.put("nb_hive_deployments", 1)
 variables.put("hive_pa_node_name", pa_node_name)
 variables.put("hdfs_namenode_container_name", hdfs_namenode_container_name)
 variables.put("hive_container_ip", hive_container_ip.toString())
 resultMap.put("hiveservice_instance_id", service_instance_id)
+resultMap.put("hiveservice_jdbc_url", "trydev2.activeeon.com:" + hiveServer2_port)
```
Comment (fviale) [inline Hive/resources/catalog/Hive.xml:217]: same remark

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml [hunk 6]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="urn:proactive:jobdescriptor:3.14" xsi:schemaLocation="urn:proactive:jobdescriptor:3.14 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.14/schedulerjob.xsd"  name="Connect_To_Hive_Cluster_With_Beeline" projectName="13. Hive_connectors"  priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+  <variables>
+    <variable name="username" value="admin"  description="The service instance id of the targeted Hive."  advanced="false" hidden="false"/>
+    <variable name="password" value="admin"    advanced="false" hidden="false"/>
+    <variable name="cluster_ip" value="trydev2.activeeon.com"    advanced="false" hidden="false"/>
```
Comment (fviale) [inline HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml:8]: please set a better default value

File: Hive/resources/catalog/Hive.xml (hunk 4 above)
Comment (rsaidane) [reply Hive/resources/catalog/Hive.xml:211]: but to access it from an external cluster with beeline and jdbc we can't use localhost

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml [hunk 7]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="urn:proactive:jobdescriptor:3.14" xsi:schemaLocation="urn:proactive:jobdescriptor:3.14 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.14/schedulerjob.xsd"  name="Connect_To_Hive_Cluster_With_Beeline" projectName="13. Hive_connectors"  priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+  <variables>
+    <variable name="username" value="admin"  description="The service instance id of the targeted Hive."  advanced="false" hidden="false"/>
+    <variable name="password" value="admin"    advanced="false" hidden="false"/>
```
Comment (fviale) [inline HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml:7]: please add missing variable descriptions

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    <variable name="cluster_ip" value="trydev2.activeeon.com"    advanced="false" hidden="false"/>
+    <variable name="cluster_port" value="42893"    advanced="false" hidden="false"/>
+  </variables>
+  <description>
```
Comment (fviale) [inline HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml:11]: improve the description similarly as above

File: Hive/resources/catalog/Hive.xml (hunk 4 above)
Comment (fviale) [reply Hive/resources/catalog/Hive.xml:211]: nor can we use trydev2.activeeon.com if we run the hive service on demo.activeeon.com

a workflow should never contain the hostname of the test platform, ever.

**Review by mbenguig — COMMENTED** [review]
Will be fine when you will consider all Fabien's comments!

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml [hunk 9]
```diff
@@ -0,0 +1,129 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="urn:proactive:jobdescriptor:3.14" xsi:schemaLocation="urn:proactive:jobdescriptor:3.14 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.14/schedulerjob.xsd"  name="Connect_To_Hive_Cluster_With_Beeline" projectName="13. Hive_connectors"  priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+  <variables>
+    <variable name="username" value="admin" model="" description="HIVE cluster username" group="" advanced="false" hidden="false"/>
+    <variable name="password" value="Abc!147" model="" description="HIVE cluster password " group="" advanced="false" hidden="false"/>
```
Comment (fviale) [inline HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml:7]: the password should have model PA:HIDDEN so it does not appear as clean test, please do so for all other worklows which involve passwords

File: Hive/resources/catalog/Docker_Swarm_HDFS_YARN_Hive_Service.xml [hunk 10]
```diff
@@ -1,11 +1,14 @@
...
   <variables>
     <variable name="node_source_name" value="" model="PA:MODEL_FROM_URL(${PA_SCHEDULER_REST_PUBLIC_URL}/rm/model/nodesources)" description="If not empty, the workflow tasks will be run only on nodes belonging to the specified node source."  advanced="false" hidden="false"/>
     <variable name="nb_yarn_hdfs_workers" value="3" model="PA:NOT_EMPTY_STRING" description="The number of YARN workers and HDFS datanodes."  advanced="false" hidden="false"/>
     <variable name="INSTANCE_NAME" value="docker-swarm-hive-$PA_JOB_ID" model="PA:NOT_EMPTY_STRING" description="The service instance name."  advanced="false" hidden="false"/>
+    <variable name="username" value="admin"  description="Set username and password to access hive cluster using the beeline and jdbc driver"  advanced="true" hidden="false"/>
+    <variable name="password" value="admin"  description="Set username and password to access hive cluster using the beeline and jdbc driver"  advanced="true" hidden="false"/>
```
Comment (fviale) [inline Hive/resources/catalog/Docker_Swarm_HDFS_YARN_Hive_Service.xml:11]: please add PA:HIDDEN in model

here the default password makes sense (but should be encrypted by hidden model)

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml (hunk 9 above)
Comment (rsaidane) [reply HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml:7]: ok I will remove it from this workflow

File: Hive/resources/catalog/Hive.xml [hunk 11]
```diff
@@ -1,11 +1,13 @@
...
-        <variable name="yarn_service_instance_id" value="xx" model="PA:NOT_EMPTY_STRING" description="This variable must be set to the targeted YARN service instance id." advanced="false" hidden="false"/>
+        <variable name="INSTANCE_NAME" value="hive-$PA_JOB_ID" model="PA:NOT_EMPTY_STRING" description="The name of the service to be deployed."  advanced="false" hidden="false"/>
+        <variable name="hdfs_service_instance_id" value="xx" model="PA:NOT_EMPTY_STRING" description="This variable must be set to the targeted HDFS service instance id."  advanced="false" hidden="false"/>
+        <variable name="yarn_service_instance_id" value="xx" model="PA:NOT_EMPTY_STRING" description="This variable must be set to the targeted YARN service instance id."  advanced="false" hidden="false"/>
+        <variable name="username" value="admin"  description="Set username and password to access hive cluster using the beeline and jdbc driver"  advanced="true" hidden="false"/>
+        <variable name="password" value="Abc!147" model="" description="Set username and password to access hive cluster using the beeline and jdbc driver" group="" advanced="true" hidden="false"/>
```
Comment (fviale) [inline Hive/resources/catalog/Hive.xml:10]: same remarks as above.

Regarding the default password values for both, there is a risk that the user will deploy the service without changing the password, and will not be able to guess it, you could mention in the description to change the default value.

Something like:

Password that will be used to access the hive cluster. Default value 'pwd' should be changed.

File: HbaseOrchestration/resources/catalog/Phoenix_Hbase_Sql_Queries.xml (hunk 1 above)
Comment (mbenguig) [reply HbaseOrchestration/resources/catalog/Phoenix_Hbase_Sql_Queries.xml:122]: @rsaidane could you follow @fviale recommandation ?

File: Hive/resources/catalog/Hive.xml (hunk 4 above)
Comment (mbenguig) [reply Hive/resources/catalog/Hive.xml]: It seems to be done.

File: Hive/resources/catalog/Hive.xml (hunk 5 above)
Comment (mbenguig) [reply Hive/resources/catalog/Hive.xml]: Done.

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml (hunk 6 above)
Comment (mbenguig) [reply HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml]: Done.

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml (hunk 7 above)
Comment (mbenguig) [reply HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml]: Done.

File: HbaseOrchestration/resources/catalog/Run_HBase_Shell_Commands.xml [hunk 12]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
+
 <job
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-        xmlns="urn:proactive:jobdescriptor:3.13" xsi:schemaLocation="urn:proactive:jobdescriptor:3.13 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.13/schedulerjob.xsd"  name="Run_Hbase_shell_commands" projectName="11. HBase Workflows" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+        xmlns="urn:proactive:jobdescriptor:3.14" xsi:schemaLocation="urn:proactive:jobdescriptor:3.14 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.14/schedulerjob.xsd"  name="Run_HBase_Shell_Commands" projectName="11. HBase Workflows"  priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
     <variables>
         <variable name="HBASE_SERVICE_INSTANCE_ID" value="xx" model="PA:NOT_EMPTY_STRING" description="instance id"  advanced="false" hidden="false"/>
     </variables>
```
[the remark this answers is not in the document]
Comment (mbenguig) [reply HbaseOrchestration/resources/catalog/Run_HBase_Shell_Commands.xml:8]: Done.

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml (hunk 8 above)
Comment (mbenguig) [reply HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml:11]: Done.

File: HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml (hunk 9 above)
Comment (mbenguig) [reply HiveOrchestration/resources/catalog/Connect_To_Hive_Cluster_With_Beeline.xml]: @rsaidane Not Done for some of them. @fviale

File: Hive/resources/catalog/Docker_Swarm_HDFS_YARN_Hive_Service.xml (hunk 10 above)
Comment (mbenguig) [reply Hive/resources/catalog/Docker_Swarm_HDFS_YARN_Hive_Service.xml]: @rsaidane You forgot to set the password param as hidden @fviale

File: Hive/resources/catalog/Hive.xml (hunk 11 above)
Comment (mbenguig) [reply Hive/resources/catalog/Hive.xml]: @rsaidane TODO  @fviale

**Review by fviale — APPROVED** [review]

**Pull request merged**